raster/r.texture/main.c

```diff
@@ -307,7 +312,20 @@ int main(int argc, char *argv[])
     out_set.flag_null = flag.null;
     out_set.flag_ind = flag.ind;
 
-    execute_texture(data, &dim, measure_menu, measure_idx, &out_set);
+    threads = atoi(parm.nproc->answer);
+#if defined(_OPENMP)
+    /* Set the number of threads */
+    omp_set_num_threads(threads);
+    if (threads > 1)
+        G_message(_("Using %d threads for parallel computing."), threads);
+#else
+    if (threads > 1) {
+        G_warning(_("GRASS GIS is not compiled with OpenMP support, parallel "
+                    "computation is disabled."));
+        threads = 1;
+    }
+#endif
+    execute_texture(data, &dim, measure_menu, measure_idx, &out_set, threads);
```

Comment:
@cyliang368 Since the PR was merged, and later on a coverity scan was made, it created a new defect here.

Do you see if you're able to fix it in a new small PR? I don't completely see yet how it can divide by zero through a modulo.

```

** CID 1548584:  Integer handling issues  (DIVIDE_BY_ZERO)


________________________________________________________________________________________________________
*** CID 1548584:  Integer handling issues  (DIVIDE_BY_ZERO)
/raster/r.texture/main.c: 328 in main()
322         if (threads > 1) {
323             G_warning(_("GRASS GIS is not compiled with OpenMP support, parallel "
324                         "computation is disabled."));
325             threads = 1;
326         }
327     #endif
   CID 1548584:  Integer handling issues  (DIVIDE_BY_ZERO)
   In function call "execute_texture", modulo by expression "threads" which may be zero has undefined behavior.
328         execute_texture(data, &dim, measure_menu, measure_idx, &out_set, threads);
329     
330         for (i = 0; i < dim.n_outputs; i++) {
331             Rast_close(outfd[i]);
332             Rast_short_history(mapname[i], "raster", &history);
333             Rast_command_history(&history);

```

Reply:
I don't know when DIVIDE_BY_ZERO will happen, but I am sure the number of threads shouldn't be 0 or negative. I just submitted a new PR (#3917) to fix it.